src/test/components/MenuButtons.test.js

```diff
@@ -5,69 +5,195 @@
+        shareWithUrlsButton.parentElement.nextElementSibling !== undefined &&
+        shareWithUrlsButton.parentElement.nextElementSibling !== null
+      ) {
+        return shareWithUrlsButton.parentElement.nextElementSibling.getElementsByClassName(
+          'arrowPanelOkButton'
+        )[0];
+      }
+      return null;
+    };
 
-    store.dispatch(doneSymbolicating());
-    expect(Array.from(container.children)).toMatchSnapshot();
-  });
+    const getMenuButtonsContainer = () =>
+      getByTestId('menuButtonsCompositeButtonContainer');
+    const getPermalinkButton = () => getByValue('Permalink');
 
-  it('renders the MenuButtons buttons with profile.meta.networkURLsRemoved set to undefined', () => {
-    const profile = processProfile(createGeckoProfileWithJsTimings());
-    profile.meta.networkURLsRemoved = undefined;
-    const store = storeWithProfile(profile);
-    store.dispatch(startSymbolicating());
+    return {
+      store,
+      ...renderResult,
+      getShareButton,
+      getInnerShareButton,
+      getShareWithUrlsButton,
+      getInnerShareWithUrlsButton,
+      getMenuButtonsContainer,
+      getPermalinkButton,
+    };
+  }
 
-    const { container } = render(
-      <Provider store={store}>
-        <MenuButtons />
-      </Provider>
-    );
+  describe('share and save buttons', function() {
+    // Mock hash
+    const hash = 'c5e53f9ab6aecef926d4be68c84f2de550e2ac2f';
+
+    beforeAll(function() {
+      if ((window: any).TextEncoder) {
+        throw new Error('A TextEncoder was already on the window object.');
+      }
+      (window: any).TextEncoder = TextEncoder;
+    });
+
+    afterAll(async function() {
+      delete (window: any).TextEncoder;
+    });
+
+    beforeEach(function() {
+      // Flow doesn't know uploadBinaryProfileData is a jest mock.
+      (uploadBinaryProfileData: any).mockImplementation(() =>
+        Promise.resolve(hash)
+      );
+      // Flow doesn't know sha1 is a jest mock.
+      (sha1: any).mockImplementation((_data: Uint8Array) =>
+        Promise.resolve(hash)
+      );
+      // Flow doesn't know shortenUrl is a jest mock.
+      (shortenUrl: any).mockImplementation(() =>
+        Promise.resolve('https://perf-html.io/')
+      );
+    });
+
+    it('matches the snapshot when starting to symbolicate', () => {
+      const { store, container } = setup();
+      store.dispatch(startSymbolicating());
+
+      // MenuButtons is rendering a fragment with several children. We need to
+      // check all children to assess that the component renders properly.
+      expect(Array.from(container.children)).toMatchSnapshot();
+    });
+
+    it('matches the snapshot when done symbolicating', () => {
+      const { store, container } = setup();
+      store.dispatch(startSymbolicating());
+      store.dispatch(doneSymbolicating());
+      expect(Array.from(container.children)).toMatchSnapshot();
+    });
+
+    it('should share the profile when Share button is clicked', async () => {
```

Comment:
Optional: when jest tests fail, they only list the `it` block that fails. This test is quite large, and has many potential sources of failure. It's much easier to debug if the `it` blocks are more granular. Looking at this test, I would think it could be broken down into:

* "Sharing without URLs" classList changes
* "Sharing without URLs" `ProfileViewSelectors.getProfileSharingStatus` checks
* "Sharing with URLs" classList changes
* "Sharing with URLs" `ProfileViewSelectors.getProfileSharingStatus` checks

Each `it` block would then run the `setup` function again, and run the `fireEvent`.

Reply:
That makes sense to make it more granular. But the reason that I wanted to do it was to test the secondary sharing as well. Otherwise, with smaller tests cases, it's not possible to test secondary sharing button. I think we can also add more granular ones on top of that test and remove some things as well but I I'm just gonna land it that way and we can do these on a followup patch.